backend/apps/common/views.py

```
@@ -4,6 +4,7 @@
 
 
 class GetHealth(APIView):
-    """ Simple health-check endpoint to check health. """
+    """Simple health-check endpoint to check health."""
+
     def get(self, request):
-        return Response({'status':'ok'}, status=status.HTTP_200_OK)
\ No newline at end of file
+        return Response({'status': 'ok'}, status=status.HTTP_200_OK)
```

Health endpoint does not check database connectivity. Verify DB connection for a more robust health check.
```
from django.db import connections
connections['default'].cursor()  # raises error if not connected
return Response({'status': 'ok'}, status=status.HTTP_200_OK)
```

---

Django connects to the database on its own during middleware processing, so any request will fail if the database becomes unavailable

---

Thanks - good point, you’re half-right. Django *will* error on normal requests if the DB is unavailable **but** that behavior is not sufficient for robust liveness/readiness checks (and for K8s readiness probes / uptime dashboards). We still need an explicit, lightweight DB check for *health* endpoints and probe semantics.

Why explicit DB checks are needed:
- Kubernetes readiness should fail fast when DB is unreachable so traffic is not routed to a pod that can’t serve DB-backed endpoints. Relying on “Django will fail when it tries to hit the DB” is reactive and can route traffic to a pod that will immediately error for real users.
- Health dashboards / uptime monitoring expect a deterministic answer: can the app reach its DB right now? That is different from whether some later middleware will raise.
- Controlled behavior - we can return 503 with a meaningful message rather than an unhandled 500 with stack traces.
- Avoid noisy errors: the health check can be rate-limited / cached (TTL) so we don’t hammer the DB with probe traffic.

---

You are right, i apologize. I have added an explicit db connection check